examples/vision/attention_mil_classification.py

```diff
@@ -0,0 +1,575 @@
+    input_data = np.divide(input_data, 255.0)
+
+    # Count positive samples.
+    count = 0
+
+    # Take out the filter for specific class.
+    filter_class = np.where(input_labels == pos_class)[0]
+
+    # Assign new variables consisting of this class.
+    data_pos_class = input_data[filter_class]
+    labels_pos_class = input_labels[filter_class]
+
+    # From overall data, remove this class.
+    data_neg_classes = np.delete(input_data, filter_class, 0)
+    labels_neg_classes = np.delete(input_labels, filter_class, 0)
+
+    # Merge both inputs and labels to each another.
+    data = np.concatenate([data_pos_class, data_neg_classes], axis=0)
+    labels = np.concatenate([labels_pos_class, labels_neg_classes], axis=0)
+
+    # Data are ordered in such a way: [pos_class... neg_classes].
+    # Shuffle the data randomly.
+    order = np.arange(len(data))
+    np.random.shuffle(order)
+    data = data[order]
+    labels = labels[order]
+
+    for _ in range(bag_count):
+
+        # Pick a fixed size random subset of samples.
+        index = np.random.choice(data.shape[0], instance_count, replace=False)
+        instances_data = data[index]
+        instances_labels = labels[index]
+
+        # By default, all bags are labelled as 0.
+        bag_label = 0
+
+        # Check if there is at least a positive class in the bag.
+        if pos_class in instances_labels:
+
+            # Positive bag will be labelled as 1.
+            bag_label = 1
+
+            # Increment count by 1.
+            count += 1
+
+        bags.append(instances_data)
+        bag_labels.append(np.array([bag_label]))
+
+    print(f"Positive bags: {count}")
+    print(f"Negative bags: {bag_count - count}")
+
+    return (list(np.swapaxes(bags, 0, 1)), np.array(bag_labels))
+
+
+# Load desired data.
+(x_train, y_train), (x_val, y_val) = keras.datasets.mnist.load_data()
+
+# Create training data.
+train_data, train_labels = create_bags(x_train, y_train, POS_CLASS, BAG_COUNT, BAG_SIZE)
+
+# Create validation data.
+val_data, val_labels = create_bags(x_val, y_val, POS_CLASS, VAL_BAG_COUNT, BAG_SIZE)
+
+"""
+# Create and train neural networks
+We will now build the attention layer, prepare some utilities, build and train the
+entire model.
+
+## Attention operator implementation
+
+The output size of this layer is decided by the size of a single bag.
+
+The attention mechanism uses a weighted average of instances in a bag, in which the sum
+of the weights must equal to 1 (invariant of the bag size).
+
+The weight matrices (parameters) are **w** and **v**. To include positive and negative
+values, hyperbolic tangent element-wise non-linearity is utilized.
+
+**Gated attention mechanism** can be used to deal with complex relations. Another weight
+matrix, **u**, is added to the computation.
+Sigmoid non-linearity is used to overcome approximately linear behavior for *x* ∈ [−1, 1]
+by hyperbolic tangent non-linearity.
+"""
+
+
+class AttentionLayer(layers.Layer):
```

Comment:
How does this layer differ from available Attention layers here? https://keras.io/api/layers/attention_layers/

Reply:
I renamed the class in my latest commit to `MILAttentionLayer` to make it clearer. This layer is fundamentally different from the existing Attention layers. The example here is exploiting MIL problem using attention scores but the existing ones address other issues.